server/sonar-server/src/main/java/org/sonar/server/computation/step/ComputeQProfileMeasureStep.java

```diff
@@ -74,19 +77,36 @@ public QProfiles createForAny(Component component) {
     }
 
     @Override
+    public void visitFile(Component file, Path<QProfiles> path) {
+      String languageKey = file.getFileAttributes().getLanguageKey();
+      if (languageKey == null) {
+        // No qprofile for unknown languages
+        return;
+      }
+      if (!analysisMetadataHolder.getQProfilesByLanguage().containsKey(languageKey)) {
+        // Should never occurs except if scanner report is buggy (ie list of qprofile is not consistent with file languages)
+        // TODO Should we fail fast, log?
+        // For now silently ignore
+        return;
+      }
+      path.parent().add(analysisMetadataHolder.getQProfilesByLanguage().get(languageKey));
+    }
+
+    @Override
+    public void visitDirectory(Component directory, org.sonar.server.computation.component.PathAwareVisitor.Path<QProfiles> path) {
```

Comment:
Remove full package of org.sonar.server.computation.component.PathAwareVisitor.Path